pennylane/templates/layers/particle_conserving_u1.py

```diff
@@ -0,0 +1,267 @@
+    :math:`U_A(\phi)`
+
+    .. math::
+
+        U_A(\phi) = \left(\begin{array}{cc} 0 & e^{-i\phi} \\ e^{-i\phi} & 0 \\ \end{array}\right)
+
+    in terms of the quantum operations supported by PennyLane.
+
+    :math:`U_A(\phi)` is used in `arXiv:1805.04340 <https://arxiv.org/abs/1805.04340>`_),
+    to define two-qubit exchange gates required to build particle-conserving
+    VQE ansatze for Quantum Chemistry simulations. See :func:`~.ParticleConservingU1`.
+
+    :math:`U_A(\phi)` is expressed in terms of ``PhaseShift``, ``Rot`` and ``PauliZ`` operations
+    :math:`U_A(\phi) = R_\phi(-2\phi) R(-\phi, \pi, \phi) \sigma_z`.
+
+    Args:
+        phi (float): angle :math:`\phi` defining the unitary :math:`U_A(\phi)`
+        wires (list[Wires]): the wires ``n`` and ``m`` the circuit acts on
+    """
+
+    n, m = wires
+
+    CZ(wires=wires)
+    CRot(-phi, np.pi, phi, wires=wires)
+
+    # decomposition of C-PhaseShift(2*phi) gate
+    PhaseShift(-phi, wires=m)
+    CNOT(wires=wires)
+    PhaseShift(phi, wires=m)
+    CNOT(wires=wires)
+    PhaseShift(-phi, wires=n)
+
+
+def u1_ex_gate(phi, theta, wires=None):
+    r"""Implement the two-qubit exchange gate :math:`U_{1,\mathrm{ex}}` proposed
+    in `arXiv:1805.04340 <https://arxiv.org/abs/1805.04340>`_ to build
+    a hardware-efficient particle-conserving VQE ansatz for Quantum Chemistry
+    simulations.
+
+    Args:
+        phi (float): angle entering the unitary :math:`U_A(\phi)`
+        theta (float): angle entering the rotation :math:`R(0, 2\theta, 0)`
+        wires (list[Wires]): the two wires ``n`` and ``m`` the circuit acts on
+    """
+
+    # C-UA(phi)
+    decompose_ua(phi, wires=wires)
+
+    qml.CZ(wires=wires[::-1])
+    qml.CRot(0, 2 * theta, 0, wires=wires[::-1])
+
+    # C-UA(-phi)
+    decompose_ua(-phi, wires=wires)
+
+
+@template
+def ParticleConservingU1(weights, wires, init_state=None):
+    r"""Implements the heuristic VQE ansatz for Quantum Chemistry simulations using the
+    particle-conserving gate :math:`U_{1,\mathrm{ex}}` proposed by Barkoutsos *et al.* in
+    `arXiv:1805.04340 <https://arxiv.org/abs/1805.04340>`_ .
+
+    This template prepares :math:`N`-qubit trial states by applying :math:`D` layers of the
+    entangler block :math:`U_\mathrm{ent}(\vec{\phi}, \vec{\theta})` to the Hartree-Fock
+    state
+
+    .. math::
+
+        \vert \Psi(\vec{\phi}, \vec{\theta}) \rangle = \hat{U}^{(D)}_\mathrm{ent}(\vec{\phi}_D,
+        \vec{\theta}_D) \dots \hat{U}^{(2)}_\mathrm{ent}(\vec{\phi}_2, \vec{\theta}_2)
+        \hat{U}^{(1)}_\mathrm{ent}(\vec{\phi}_1, \vec{\theta}_1) \vert \mathrm{HF}\rangle.
+
+    The circuit implementing the entangler blocks is shown in the figure below.
+
+    |
+
+    .. figure:: ../../_static/templates/layers/particle_conserving_u1.png
+        :align: center
+        :width: 50%
+        :target: javascript:void(0);
+
+    |
+
+    The repeated units across several qubits are shown in dotted boxes. Each layer
```

Comment:
I found this hard to understand. Are you trying to say that the border type indicates the same unit? What is a unit? How does it relate to U_ent?

Reply:
Oh, U_ent is the entire block in the figure. To make that super easy to grasp, do you want to add "U_ent" to the picture?

Reply:
> I found this hard to understand. Are you trying to say that the border type indicates the same unit? What is a unit? How does it relate to U_ent?

The repeated units refers to the exchange gates acting on any qubit labelled as |.>. @mariaschuld this is because the figure should be general for any number of qubits and electrons (the 1s).